### PR TITLE
feat: add progress bar to lower jaw

### LINKED ISSUE #50386: Inner progress bar will not be visible when completion modal is re-opened
### Describe the Issue

Inner progres bar will not be showed when the camper re-open the completion modal after they close the modal. 

### Affected Page

https://www.freecodecamp.org/learn/college-algebra-with-python/learn-ratios-and-proportions/introduction-to-college-algebra-with-python

### Steps to Reproduce

1. Go to any challenges which has the button to show the completion modal after solving the challenge.
2. Click on the button (such as "Run the Tests(Ctrl + Enter)" of JavaScript Algorithms and Data Structures challenges, or "Check your answer" of College Algebra with Python challenge) after solving the challenge.
3. The Inner progress bar will be showed with animation in pop up completion modal, and under the progress bar, completion percentage will be showed by text as "xx% complete"
4. Close the completion modal once. (The camper may do this for reveiw the challenge)
5. Then re-open the modal by clicking the button, the inner progress bar will not be showed, though completion percentage text will also be there and convey the camper the proper percentage.

### Expected behavior

The inner progress bar will be showed when the completion modal is re-opened.
(I'm not sure this should be with animation or not)

### Screenshots

inner_progress_bar_is_not_showed

### System

- Device: iPhone SE and Laptop(64 bit)
- OS: iOS 15.7.5 and Windows 11
- Browser: Chrome, Firefox, Edge, Brave for Laptop, or Chrome on iPhoneSE
- Version: Chrome => 113.0.5672.93 (Official Build), Firefox => 113.0, Edge => 113.0.1774.42 (Official Build)

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by Sembauke**
- feat add lower jaw back
- fix: merge conflicts lower jaw and progress bar

Checklist:

- [x] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [x] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/#/how-to-open-a-pull-request).
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

Closes #50386

### CODE REVIEW

**Comment by ahmadabdolsaheb** [comment]
Thanks for working on this. Let us make sure we give credit to @Ismailtlem before merging.

File: client/src/components/ProgressBar/progress-bar.tsx [hunk 1]
```diff
@@ -84,4 +82,4 @@ function ProgressBar({
 
 ProgressBar.displayName = 'ProgressBar';
 
-export default connect(mapStateToProps)(withTranslation()(ProgressBar));
+export default ProgressBar;
```
Comment (Sboonny) [inline client/src/components/ProgressBar/progress-bar.tsx:85]: If I am not missing something, losing the connection to redux store forces you to add props, which you did, but you need to add it every time the component is added.

you can:
- connect it back to redux store, and remove the props.
- add the props to every place that use progress bar.

Comment (Sembauke) [reply client/src/components/ProgressBar/progress-bar.tsx:85]: Hey @Sboonny,

Thanks for reaching out, please follow this comment:

https://github.com/freeCodeCamp/freeCodeCamp/pull/49395#issuecomment-1499989523

**Comment by Sembauke** [comment]
> @Sembauke In the `LowerJawStatus` function, can we tack on the percentage complete message in sr-only text immediately after the `congratulationText`? So that screen reader users will hear the congrats message and then "N% complete."

I will look into it!

**Comment by ahmadabdolsaheb** [comment]
So here are a few issues that I came across with the progress bar.
- To apply the animation on every step, the applyAnimation and percent variables should be set to states and updated accordingly.
- Since the progress bar is connected to redux, completedPercent updates after each challenge completion. As a result, window.setInterval runs without progress bar being on the screen. One solution could be to check if the component is on the view port before setting the interval. The might be a better a better solution, so I am open to suggestions.

**Comment by ahmadabdolsaheb** [comment]
The progress bar should be working as expected.
The component could be simplified further, but seems out of scope for this pr.

**Comment by ahmadabdolsaheb** [comment]
@Sembauke feel free to add a couple of e2e tests here and open the pr for review.

File: cypress/e2e/default/learn/challenges/progress-bar.ts [hunk 2]
```diff
@@ -22,4 +22,23 @@ describe('progress bar', () => {
...
+      cy.get(`${'.react-monaco-editor-container'} textarea`, { timeout: 16000 })
+        .click()
+        .focused()
+        .type('{ctrl}a')
+        .clear()
+        .type(`<h1>CatPhotoApp</h1>\n<h2>Cat Photos</h2>`);
+      cy.contains('Check Your Code (Ctrl + Enter)').click({ force: true });
+      cy.contains('Submit and go to next challenge');
+      cy.get('.progress-bar-container').contains('1% complete');
+    }
```
[the remark this answers is not in the document]
Comment (Sembauke) [reply cypress/e2e/default/learn/challenges/progress-bar.ts:42]: Yes, the new one tests the progressbar in the jaw and the other in the completion modal

**Review by naomi-lgbt — APPROVED** [review]
Looks good to me.

**Comment by bbsmooth** [comment]
> can we tack on the percentage complete message in sr-only text immediately after the congratulationText? 

If we want to push this through now without this, that's fine. I realize this is not a simple update, so I'll fix it after the fact. I just wanted to bring it up again because it's in my nature to be a pest.

**Comment by Sembauke** [comment]
Sorry this wasn't ready for review. There still seem to be some issues.

**Comment by ahmadabdolsaheb** [comment]
Thank you @Sembauke for your hard work trying to get this one in. I tested this pr, but didn't come across an issue. I also had a chance to look into the inconsistency that you mentioned in chat about the percentage turning to 1 after the tests run on the 3rd challenge. Could you please double check if the issue still exist and if there are other issues that you want me to look into?

**Review by ahmaxed — APPROVED** [review]

**Comment by ahmadabdolsaheb** [comment]
Thank you @Ismailtlem and @Sembauke for your hard work. We have a lower jaw progress bar now :)

**Pull request merged**